File: frontend/src/utils/assetDisplay.ts
import { ASSET_KINDS } from '../components/form/documentFormConstants';
import type { AssetDto } from '../types';

/** تسمية نوع الأصل للعرض داخل بطاقته (عقار / مركبة / متجر / كفالة رواتب / متجر غير مسجل). */
export function assetKindLabel(kind: string | undefined): string {
  switch (kind) {
    case ASSET_KINDS.vehicle:
      return 'مركبة';
    case ASSET_KINDS.shop:
      return 'متجر';
    case ASSET_KINDS.salaryGuarantee:
      return 'كفالة رواتب';
    case ASSET_KINDS.unregisteredShop:
      return 'متجر غير مسجل';
    default:
      return 'عقار';
  }
}

/**
 * تسمية قراءة للأصل في قوائم الاختيار والعرض (مثال: «مركبة سيارة — لوحة 123»):
 * تُستعمل في «منفذ جبريا» وفي بطاقات الأموال. قيمة النوع تُقدَّم قبل التفاصيل الخاصة به.
 */
export function assetDisplayName(a: AssetDto & { id?: number }): string {
  if (a.assetKind === ASSET_KINDS.vehicle) {
    const type = (a.vehicleType ?? '').trim();
    const plate = (a.plateNumber ?? '').trim();
    if (type && plate) return `مركبة ${type} — لوحة ${plate}`;
    if (plate) return `مركبة لوحة ${plate}`;
    return type ? `مركبة ${type}` : `مركبة ${a.id ?? ''}`.trim();
  }
  if (a.assetKind === ASSET_KINDS.shop) {
    const reg = (a.registerNumber ?? '').trim();
    if (reg) return `متجر سجل رقم ${reg}`;
    return (a.shopDescription ?? '').trim() || `متجر ${a.id ?? ''}`.trim();
  }
  if (a.assetKind === ASSET_KINDS.salaryGuarantee) {
    const entity = (a.publicEntity ?? '').trim();
    return entity ? `كفالة رواتب — ${entity}` : 'كفالة رواتب';
  }
  if (a.assetKind === ASSET_KINDS.unregisteredShop) {
    const license = (a.licenseNumber ?? '').trim();
    return license ? `متجر غير مسجل ترخيص رقم ${license}` : `متجر غير مسجل ${a.id ?? ''}`.trim();
  }
  const property = (a.property ?? '').trim();
  if (property) return property;
  return a.propertyNumber ? `عقار رقم ${a.propertyNumber}` : `عقار ${a.id ?? ''}`.trim();
}

File: backend/src/DocGenerator.Application/Common/AssetDisplay.cs
using DocGenerator.Domain.Entities;
using DocGenerator.Domain.Enums;

namespace DocGenerator.Application.Common;

/// <summary>
/// تسمية قراءة لأصل (عقار/مركبة/متجر/متجر غير مسجل) — مصدر الحقيقة الوحيد لوصف الأصل
/// في النصوص والإنابات (مثال: «عقار رقم 77 — المزة»). يُستخدم لتقط لقطة أصول الإنابة.
/// </summary>
public static class AssetDisplay
{
    public static string Label(Asset a)
    {
        var kind = a.AssetKind;
        if (kind == AssetKindCatalog.RealEstate)
        {
            var property = (a.Property ?? string.Empty).Trim();
            if (property.Length > 0) return property;
            return string.IsNullOrWhiteSpace(a.PropertyNumber) ? $"عقار {a.Id}" : $"عقار رقم {a.PropertyNumber}";
        }
        if (kind == AssetKindCatalog.Vehicle)
        {
            var type = (a.VehicleType ?? string.Empty).Trim();
            var plate = (a.PlateNumber ?? string.Empty).Trim();
            if (type.Length > 0 && plate.Length > 0) return $"مركبة {type} — لوحة {plate}";
            if (plate.Length > 0) return $"مركبة لوحة {plate}";
            return type.Length > 0 ? $"مركبة {type}" : $"مركبة {a.Id}";
        }
        if (kind == AssetKindCatalog.Shop)
        {
            var reg = (a.RegisterNumber ?? string.Empty).Trim();
            if (reg.Length > 0) return $"متجر سجل رقم {reg}";
            return string.IsNullOrWhiteSpace(a.ShopDescription) ? $"متجر {a.Id}" : a.ShopDescription.Trim();
        }
        if (kind == AssetKindCatalog.UnregisteredShop)
        {
            var license = (a.LicenseNumber ?? string.Empty).Trim();
            if (license.Length > 0) return $"متجر غير مسجل ترخيص رقم {license}";
            return $"متجر غير مسجل {a.Id}";
        }
        return string.Empty;
    }
}
